src/spikeinterface/extractors/neoextractors/neobaseextractor.py

```diff
@@ -330,44 +569,36 @@ def __init__(
         neo_reader,
         block_index,
         segment_index,
-        use_natural_unit_ids,
         t_start,
         sampling_frequency,
-        handle_spike_frame_directly,
+        neo_returns_frames,
     ):
         BaseSortingSegment.__init__(self)
         self.neo_reader = neo_reader
         self.segment_index = segment_index
         self.block_index = block_index
-        self.use_natural_unit_ids = use_natural_unit_ids
         self._t_start = t_start
         self._sampling_frequency = sampling_frequency
-        self._natural_ids = None
-        self.handle_spike_frame_directly = handle_spike_frame_directly
+        self.neo_returns_frames = neo_returns_frames
 
-    def get_natural_ids(self):
-        if self._natural_ids is None:
-            self._natural_ids = list(self._parent_extractor().neo_reader.header["spike_channels"]["id"])
-        return self._natural_ids
```

Comment:
I guess the ida of this function was to delay construction of the dict to make the init faster.
But this is a details. Lets keep this change.